bluetooth/characteristic/writeValue/buffer-is-detached.https.html

```diff
@@ -0,0 +1,36 @@
+<!DOCTYPE html>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/resources/testdriver.js"></script>
+<script src="/resources/testdriver-vendor.js"></script>
+<script src="/bluetooth/resources/bluetooth-helpers.js"></script>
+<script>
+'use strict';
+
+function detachBuffer(buffer) {
+  window.postMessage('', '*', [buffer]);
+}
+
+bluetooth_test(async () => {
+  let {characteristic, fake_characteristic} =
+      await getMeasurementIntervalCharacteristic();
+
+  let typed_array = Uint8Array.of(1, 2);
+  detachBuffer(typed_array.buffer);
+  try {
+    await characteristic.writeValue(typed_array);
+    assert_unreached();
+  } catch (e) {
+    assert_equals(e.code, DOMException.INVALID_STATE_ERR, e.toString());
```

Comment:
We've got `assert_throws` and `promise_rejects` for this.

Reply:
Tweaked the code to use `promise_rejects` - PTAL

Reply:
I don't see it?

Reply:
Sorry @Ms2ger I had stashed them locally to check the before/after state!

Reply:
@Ms2ger PTAL again?